### PR TITLE
Revert "Bridging perf from NeMo2 to Mbridge for certain configs (#2199)"

### PR DESCRIPTION
**Pull request by ko3n1g**
This reverts commit 5622b5c8d762e0887883e31c9c0d5a6ec97bc58e.

# What does this PR do ?

Leads to a hang on all perf scripts (only perf)

# Changelog

- Add specific line by line info of high level changes in this PR.

# GitHub Actions CI

See the [CI section](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md#-running-github-ci)in the Contributing doc for how to trigger the CI. A Nvidia developer will need to approve and trigger the CI for external contributors.

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?

If you haven't finished some of the above items you can still open "Draft" PR.

# Additional Information

- Related to # (issue)

## Release Notes

* **Breaking Changes**
  * Removed configuration fields: `check_optimizer_step_success` and `skip_sync_grad_norm_across_mp`
  * Timing log level restricted to values 0, 1, or 2
  * Removed `p2p_communicator` parameter from training function

* **New Features**
  * Per-virtual-stage data iterators for virtual pipeline parallelism
  * Dynamic per-batch FLOP counting
  * Evaluation duration and iteration tracking

* **Improvements**
  * Enhanced learning rate extraction from optimizer parameter groups
  * Streamlined fault tolerance and profiling execution

### CODE REVIEW

**Comment by ko3n1g** [comment]
confirmed via CI that this revert fixes the hang
